### DISCUSSION TITLE
Request: PolygonZone determination using object recognition

### DISCUSSION THREAD

**Discussion by pasionline** [roboflow/supervision#1244] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Description

I am currently using supervision in my thesis to analyse driving behavior in different videos and its super useful. But the PolygonZone array must be determined manually for each video.

Would it be possible to (semi-) automate this process with object recognition? By specifying an object that can be found in several places in a frame, the feature would then return the coordinates of the object from the frame and append them to an array.

### Use case

It would be very useful, for example, when determining the polygon zone, which is created on the basis of delineators. In this way, a road section can be recognized directly without having to enter an array manually.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by SkalskiP** [comment]
Hi @pasionline 👋🏻 let me convert this issue into discussion

**Comment by SkalskiP** [comment]
Hi @pasionline 👋🏻, it is possible to do it right now if you know what objects you should be looking for. Do you have a model that could detect those things?

**Comment by M-Colley** [comment]
I would be interested in this feature too. I have briefly checked whether there are dataset for guide posts but that does not seem to be the case. @SkalskiP I have seen your tutorials on fine-tuning models for new object types, do you have any estimation how many examples one would need to train e.g. YOLOv9 to detect guide posts? 

I would be open to provide such data (openly). 


Kind regards

**Comment by pasionline** [comment]
Hey @SkalskiP , I could provide now a fine-tuned YOLOv10 model which could detect those traffic delineators, if you are still interested :)

**Reply by pasionline** [reply]
Also here are the links to the Dataset:
https://universe.roboflow.com/baschenk/traffic-delineators-detection
and to the google collab:
https://colab.research.google.com/drive/1E3h0itn4fznDD1-9t1xWb2TEgy-41Zc8?usp=sharing
